[Bug]: `FilterChain.add_filter` incorrect type

### crawl4ai version

0.8.0

### Expected Behavior

You should be able to use `add_filter` to add a filter.

### Current Behavior

```
class FilterChain:
...
    def __init__(self, filters: List[URLFilter] = None):
        self.filters = tuple(filters or [])  # Immutable tuple for speed
        ...

    def add_filter(self, filter_: URLFilter) -> "FilterChain":
        """Add a filter to the chain"""
        self.filters.append(filter_)
        return self  
```
```
E   AttributeError: 'tuple' object has no attribute 'append'
```

### Is this reproducible?

Yes


### Steps to Reproduce

```bash
FilterChain([...]).add_filter([...])
```

--- a/crawl4ai/deep_crawling/filters.py
+++ b/crawl4ai/deep_crawling/filters.py
@@ -85,7 +85,7 @@ class FilterChain:
 
     def add_filter(self, filter_: URLFilter) -> "FilterChain":
         """Add a filter to the chain"""
-        self.filters.append(filter_)
+        self.filters = self.filters + (filter_,)
         return self  # Enable method chaining
 
     async def apply(self, url: str) -> bool:

